### PR TITLE
Add CreateFolder Functionality

### PR DESCRIPTION
**Pull request by unclebiguns**
Added function CreateFolder to create a folder in the catalog.  
Added call to CreateFolder function in DeployProject so that if the folder does not exist it is created and the deployment will succeed.
Added switches to enable the ability to just create a folder in the catalog and added the code path to handle that functionality

### CODE REVIEW

File: ISDeploymentCmd/deploy_project.cs [hunk 1]
```diff
@@ -8,8 +8,11 @@ public partial class CatalogProc
     {
         public int DeployProject(string folderName, string projectName, byte[] projectStream)
         {
+
             using (var connection = new SqlConnection(ConnectionString))
             {
+                int folderCreated = CreateFolder(folderName, connection);
```
Comment (tkwj) [inline ISDeploymentCmd/deploy_project.cs:14]: The automatic create here is a nice touch!

File: ISDeploymentCmd/deploy_project.cs [hunk 2]
```diff
@@ -19,10 +22,60 @@ public int DeployProject(string folderName, string projectName, byte[] projectSt
...
             }
         }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="folderName"></param>
+        /// <param name="connection"></param>
+        /// <returns></returns>
+        public int CreateFolder(string folderName, SqlConnection connection = null)
```
Comment (tkwj) [inline ISDeploymentCmd/deploy_project.cs:40]: I'm going to refactor CreateFolder into a separate file as another partial class to fit the pattern of my CatalogProcedures library.

**Review by tkwj — APPROVED** [review]

**Comment by tkwj** [comment]
Thanks for this feature!

File: ISDeploymentCmd/deploy_project.cs (hunk 2 above)
Comment (unclebiguns) [reply ISDeploymentCmd/deploy_project.cs:40]: Sounds good.

**Pull request merged**